### PR TITLE
Allow selection of entitites directly in the plot space view

### LINKED ISSUE #4828: Improve X-axis ticks spacing in timeseries space views
image

that X axis ☝🏻 is way too messy

### LINKED ISSUE #4952: Ability to select Scalar entity in TimeSeriesView
Selecting an entity currently requires using the blueprint tree. We should be able to directly click lines / points to select them.

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* depends on https://github.com/emilk/egui/pull/3920
* Fixes #4952
* Closes #4828

Wires up the new plot item ids to our hover/selection mechanism.
Moved a utility method from item_ui to ViewerContext since I didn't want to take a dependency on `re_data_ui` here (and this method had a todo about being in the wrong spot anyways 😄 )

https://github.com/rerun-io/rerun/assets/1220815/1ef61a4f-7e13-4661-90c6-fa53ea008823

DRAFT: Uses unpublished egui version!

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4959/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4959/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4959/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4959)
- [Docs preview](https://rerun.io/preview/f3f871cc9d2504b9c12a445787116a69af91f47f/docs) 
- [Examples preview](https://rerun.io/preview/f3f871cc9d2504b9c12a445787116a69af91f47f/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by nikolausWest** [comment]
Fyi: Bar charts aren't clickable

**Comment by Wumpf** [comment]
done! now also works with bar charts. Thanks for checking on it @nikolausWest :)

Now based on egui `master`, ready for review! 🚢

**Review by emilk — APPROVED** [review]
sweet

**Comment by Wumpf** [comment]
I just noticed that when you hover on a thing in the legend it actually highlights by boosting the size 🤔 
But I don't think that's a good idea for hover because you loose so much precision - it's fine on the legend since that's "far off"
Really what I'd want is the same outline rendering we have in re_Renderer driven space views....

File: Cargo.toml [hunk 1]
```diff
@@ -277,13 +277,13 @@ debug = true
...
 
-ecolor = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }      # egui master 2024-01-29
-eframe = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }      # egui master 2024-01-29
-egui = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }        # egui master 2024-01-29
-egui_extras = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" } # egui master 2024-01-29
-egui_plot = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }   # egui master 2024-01-29
-egui-wgpu = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }   # egui master 2024-01-29
-emath = { git = "https://github.com/emilk/egui.git", rev = "ab39420c2933d2e402999043375b64e7cf0ee9ed" }       # egui master 2024-01-29
+ecolor = { git = "https://github.com/emilk/egui.git", rev = "ca513ce241c6511275e92b05b2953c38fa6086e1" }      # egui Ids on plot items. 2024-01-30
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply Cargo.toml:280]: ah this used to just say the branch name. Eh whatever, not gonna torture ci with this again now when it's about to be changed again 😄

**Pull request merged**